SMLHelper/Logger.cs

```diff
@@ -1,36 +1,86 @@
 ﻿namespace SMLHelper.V2
 {
     using System;
+    using System.IO;
+
+    internal enum LogLevel
+    {
+        Debug = 0,
+        Warn = 1,
+        Error = 2,
+        Info = 3
+    }
 
     internal static class Logger
     {
-        internal static void Log(string text)
+        internal static LogLevel LoggingLevel { get; private set; }
+
+        internal static void Initialize()
+        {
+            string configPath = "./QMods/Modding Helper/loglevel.txt";
+
+            if(!File.Exists(configPath))
+            {
+                File.WriteAllText(configPath, "Warn");
+                LoggingLevel = LogLevel.Warn;
+
+                return;
+            }
+
+            string level = File.ReadAllText(configPath);
+
+            try
+            {
+                LoggingLevel = (LogLevel)Enum.Parse(typeof(LogLevel), level);
+
+                if (LoggingLevel > LogLevel.Error)
+                {
+                    LoggingLevel = LogLevel.Error;
+
+                    File.WriteAllText(configPath, "Error");
+                }
+
+                Log($"Log level set to: LogLevel.{LoggingLevel.ToString()}.", LogLevel.Info);
+            }
+            catch (Exception)
+            {
+                LoggingLevel = LogLevel.Warn;
+
+                File.WriteAllText(configPath, "Warn");
+
+                Log("Error reading log level configuration file. Defaulted to LogLevel.Warn.", LogLevel.Warn);
+            }
+        }
+
+        internal static void Log(string text, LogLevel level = LogLevel.Error)
         {
-            Console.WriteLine($"[SMLHelper] {text}");
+            if(level >= LoggingLevel)
+                Console.WriteLine($"[SMLHelper/{level.ToString()}] {text}");
```

Comment:
It would make things easier to read if we also add a new set of methods to the `Logger` class.
```CSharp
internal static void Debug(string text) => Log(text, LogLevel.Debug);  
internal static void Warn(string text) => Log(text, LogLevel.Warn);  
internal static void Error(string text) => Log(text, LogLevel.Error);  
internal static void Info(string text) => Log(text, LogLevel.Info);
```
If you also want to update the current method calls you've updated to use these methods, you're welcome to.  
For now, just having the option is enough.